cognite/client/config.py

```diff
@@ -46,6 +59,43 @@ def __init__(self) -> None:
         self.max_workers: int = 5
         self.silence_feature_preview_warnings: bool = False
 
+    def apply_settings(self, settings: dict[str, Any] | str) -> None:
+        """Apply settings to the global configuration object from a YAML/JSON string or dict.
+
+        Note:
+            All settings in the dictionary will be applied unless an invalid key is provided, a ValueError will instead be raised and no settings will be applied.
+
+        Warning:
+            This must be done before instantiating a CogniteClient for the configuration to take effect.
+
+        Args:
+            settings (dict[str, Any] | str): A dictionary or YAML/JSON string containing configuration values defined in the GlobalConfig class.
+
+        Examples:
+
+            Apply settings to the global_config from a dictionary input:
+
+                >>> from cognite.client import global_config
+                >>> settings = {
+                ...     "max_retries": 5,
+                ...     "disable_ssl": True,
+                ... }
+                >>> global_config.apply_settings(settings)
+        """
+
+        loaded = load_resource_to_dict(settings).copy()  # doing a shallow copy to avoid mutating the user input config
+
+        if not loaded.keys() <= self.__dict__.keys():
+            raise ValueError(
+                f"One or more invalid keys provided for global_config, no settings applied: {set(loaded) - set(self.__dict__)}"
+            )
+
+        if "default_client_config" in loaded:
+            if not isinstance(loaded["default_client_config"], ClientConfig):
+                loaded["default_client_config"] = ClientConfig.load(loaded["default_client_config"])
+
+        vars(self).update(loaded)
```

Comment:
You use `__dict__` directly two times, then go for `vars` at the end; I would suggest we either use dunder dict everywhere _or_ move `current_settings = vars(self)` then reference it?

(also added in a suggestion using `dict_keys.__sub__`)

```suggestion
        current_settings = vars(self)
        if not loaded.keys() <= current_settings:
            raise ValueError(
                f"One or more invalid keys provided for global_config, no settings applied: {loaded.keys() - current_settings}"
            )

        if "default_client_config" in loaded:
            if not isinstance(loaded["default_client_config"], ClientConfig):
                loaded["default_client_config"] = ClientConfig.load(loaded["default_client_config"])

        current_settings.update(loaded)
```